src/overlays/actors/ovl_Bg_Iknv_Obj/z_bg_iknv_obj.c

```diff
@@ -21,32 +31,190 @@ const ActorInit Bg_Iknv_Obj_InitVars = {
+void BgIknvObj_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgIknvObj* this = THIS;
+    CollisionHeader* colHeader = NULL;
+
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+    this->actionFunc = BgIknvObj_DoNothing;
+    switch (IKNV_OBJ_TYPE(this)) {
+        case IKNV_OBJ_WATERWHEEL: {
+            this->displayListPtr = D_06013058;
+            this->actionFunc = BgIknvObj_UpdateWaterwheel;
+            this->dyna.actor.flags |= 0x100000;
+            this->dyna.actor.flags |= 0x10;
+            return;
+        }
+        case IKNV_OBJ_RAISED_DOOR: {
+            this->displayListPtr = D_06011880;
+            BcCheck3_BgActorInit(&this->dyna, 0);
+            BgCheck_RelocateMeshHeader(&D_060119D4, &colHeader);
+            this->dyna.bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, &this->dyna, colHeader);
+            this->actionFunc = BgIknvObj_UpdateRaisedDoor;
+            this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y + 120.0f;
+            return;
+        }
+        case IKNV_OBJ_SAKON_DOOR: {
+            this->displayListPtr = D_060129C8;
+            this->actionFunc = BgIknvObj_UpdateSakonDoor;
+            BcCheck3_BgActorInit(&this->dyna, 0);
+            BgCheck_RelocateMeshHeader(&D_06012CA4, &colHeader);
+            this->dyna.bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, &this->dyna, colHeader);
+            Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+            Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+            this->dyna.actor.colChkInfo.mass = 0xFF;
+            gSaveContext.weekEventReg[0x33] &= 0xEF;
+            Actor_SetHeight(&this->dyna.actor, IREG(88));
+            return;
+        }
+    }
+
+    Actor_MarkForDeath(&this->dyna.actor);
+}
+
+void BgIknvObj_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgIknvObj* this = THIS;
+
+    if (IKNV_OBJ_TYPE(this) != IKNV_OBJ_RAISED_DOOR) {
+        if (IKNV_OBJ_TYPE(this) == IKNV_OBJ_SAKON_DOOR) {
+            Collider_DestroyCylinder(globalCtx, &this->collider);
+            gSaveContext.weekEventReg[0x33] &= 0xEF;
+        } else {
+            return;
+        }
+    }
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+s32 func_80BD7CEC(BgIknvObj* this) {
+    if (this->dyna.actor.cutscene == -1) {
+        return 1;
+    }
+
+    if (ActorCutscene_GetCurrentIndex() == this->dyna.actor.cutscene) {
+        return 1;
+    }
+
+    if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        return 1;
+    }
+
+    ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    return 0;
+}
+
+void BgIknvObj_UpdateWaterwheel(BgIknvObj* this, GlobalContext* globalCtx) {
+    if (gSaveContext.weekEventReg[0xE] & 4) {
+        this->dyna.actor.shape.rot.z += -0x64;
```

Comment:
```suggestion
        this->dyna.actor.shape.rot.z -= 0x64;
```
I know negatives can be finicky sometimes, does this work?